src/Nest/Aggregations/AggregateDictionary.cs

```diff
@@ -269,7 +269,9 @@ private IEnumerable<KeyedBucket<TKey>> GetKeyedBuckets<TKey>(IEnumerable<IBucket
 			foreach (var bucket in buckets)
 				yield return new KeyedBucket<TKey>(bucket.BackingDictionary)
 				{
-					Key = (TKey)Convert.ChangeType(bucket.Key, typeof(TKey)),
+					Key = typeof(TKey).IsEnum
```

Comment:
Factor this out into a method called by `GetKeyedBuckets<TKey>`, `GetSignificantTermsBuckets<TKey>` and `GetRareTermsBuckets<TKey>`. Maybe `GetKeyValue<TKey>(bucket.Key)` ?

Reply:
Have factored out